Circular progress (#1723)
looker-open-source/components: www/src/documentation/components/content/progress-circular.mdx

---
title: ProgressCircular
storybook: true
---

```jsx
<SpaceVertical gap="xsmall">
  <ProgressCircular />
</SpaceVertical>

> ```

Author: Going to be working on building out some docs for this

---
title: ProgressCircular
storybook: true
status: experimental
---

The `ProgressCircular` helps express an unspecified wait time or displays the length of a process

> ## Showing an unknown wait time

Reviewer: ```suggestion
## Showing a generic loading state
```

If the time a process takes is not known or detectable, use the default `ProgressCircular` behavior.

```jsx
<Space justifyContent="center">
  <ProgressCircular />
</Space>
```

> ## Indicating a known wait time

Reviewer: ```suggestion
## Rendering defined async progress
```

> If the time a process takes can be determined, you can use the `progress` prop to indicate how far along the process has progressed. The `progress` prop accepts a number from `0` to `1`, where `0` is 0% complete and `1` is 100% complete.

Reviewer: ```suggestion
If the process has known steps to completion, you can use the `progress` prop to indicate a more precise loading state. The `progress` prop accepts a number from `0` to `1`, where `0` is 0% complete and `1` is 100% complete.
```

I think we should remove reference to time from these docs, since it's not actually dealing in seconds or minutes.

> ```jsx
> ;() => {
>   const options = [
>     { value: '0', label: '0%' },
>     { value: '.25', label: '25%' },
>     { value: '.50', label: '50%' },
>     { value: '.75', label: '75%' },
>     { value: '1', label: '100%' },
>   ]
> 
>   const [progress, setProgress] = useState(options[2].value)
> 
>   return (
>     <SpaceVertical>
>       <ProgressCircular progress={progress} />
>       <Space>
>         <ButtonToggle
>           value={progress}
>           options={options}
>           onChange={setProgress}
>         />
>       </Space>
>     </SpaceVertical>
>   )
> }
> ```

Reviewer: The fixed ButtonToggle values are a nice solution! If you did still want to render a slider UI, you could just divide the value by 100 to get the decimal:

```suggestion
```jsx
;() => {
  const [progress, setProgress] = useState(50)

  return (
    <SpaceVertical>
      <ProgressCircular progress={progress / 100} />
      <Space>
        <Slider
          min={0}
          max={100}
          value={progress}
          onChange={(e) => {
            setProgress(e.target.value)
          }}
        />
      </Space>
    </SpaceVertical>
  )
}
```

Either way is fine with me though!
